flair/models/lemmatization_model.py

```diff
@@ -0,0 +1,516 @@
+            word_seq_lenght.append(len(word_seq))
+            word_seq_list.append(word_seq)
+            max_word_lenght = max(max_word_lenght, len(word_seq))
+
+            if self.use_tag:
+                tags_seq = []
+                for tag in self.tag_list:
+                    tags_seq.append(token.get_tag(tag).value)
+                tags_seq = self._item2seq(self.tag_dictionary, tags_seq)
+                tag_seq_list.append(tags_seq)
+
+            lemma_seq = self._item2seq(self.character_dictionary, token.get_tag('lemma').value)
+            lemma_seq.append(self.character_dictionary.get_idx_for_item(EOS_token))
+
+            max_lemma_lenght = max(max_lemma_lenght, len(lemma_seq))
+            lemma_seq_list.append(lemma_seq)
+
+        # Zero padding, make all sequences the same length
+        word_seq_list = self._seq_zero_padding(word_seq_list, max_word_lenght)
+        # Transpose input data.
+        word_seq_list = torch.LongTensor(word_seq_list).transpose(0, 1).to(flair.device)
+
+        if self.use_tag:
+            tag_seq_list = torch.LongTensor(tag_seq_list).transpose(0, 1).to(flair.device)
+
+        lemma_seq_list = self._seq_zero_padding(lemma_seq_list, max_lemma_lenght)
+        lemma_seq_list = torch.LongTensor(lemma_seq_list).transpose(0, 1).to(flair.device)
+
+        # Generate mask, because when calculating loss, the part through zero padding should not be included in the calculation.
+        mask = torch.BoolTensor(self._binary_matrix(lemma_seq_list)).to(flair.device)
+
+        # At the same time, update the length of the longest word encountered for the model.
+        self.longest_word_length = max(self.longest_word_length, max_word_lenght, max_lemma_lenght)
+
+        return word_seq_list, tag_seq_list, lemma_seq_list, mask, word_seq_lenght, max_lemma_lenght
+
+    def _item2seq(self, dictionary: Dictionary, input):
+        # Convert item into a sequence.
+        sequence = []
+        for item in input:
+            sequence.append(dictionary.get_idx_for_item(item))
+
+        return sequence
+
+    def _seq_zero_padding(self, seqs: list, target_length: int, fill_value=0):
+        for seq in seqs:
+            seq.extend([fill_value] * (target_length - len(seq)))
+        return seqs
+
+    def _binary_matrix(self, seqs: list):
+        matrix = []
+        for i, seq in enumerate(seqs):
+            matrix.append([])
+            for value in seq:
+                if value == 0:
+                    matrix[i].append(0)
+                else:
+                    matrix[i].append(1)
+        return matrix
+
+    def _get_state_dict(self):
+        model_state = {
+            "state_dict": self.state_dict(),
+            "hidden_size": self.hidden_size,
+            "pre_embeddings": self.pre_embeddings,
+            "character_dictionary": self.character_dictionary,
+            "n_layers": self.n_layers,
+            "tag_list": self.tag_list,
+            "tag_dictionary": self.tag_dictionary,
```

Comment:
tag_list and tag_dictionary are not set if tag_list = [] in the init. This throws an error here that you could fix with:
```python
 "tag_list": self.tag_list if "tag_list" in self.__dict__.keys() else None,
"tag_dictionary": self.tag_dictionary if "tag_dictionary" in self.__dict__.keys() else None,
```